src/components/Button/Button.tsx

```diff
@@ -33,9 +33,9 @@ export interface Props {
    */
   iconName?: string;
   /**
-   * Determines position of icon relative to button text.
-   * - **before** places icon before button text
-   * - **after** places icon after button text
+   * Determines position of icon relative to button children.
+   * - **before** places icon before button children
+   * - **after** places icon after button children
```

Comment:
I originally removed them in https://github.com/chanzuckerberg/edu-design-system/pull/889#pullrequestreview-903732508 but reverted that change because of push back.

My plan was to ask Sean how he feels about having the unreliable spacing between the icon and the text. If he's fine with it, we can leave these props and the set margin between the icon and the text. I believe that issue with the spacing is the only reason we wanted to avoid this API and let devs pass in the icons themselves (because then they can also add custom styling).

The only other downside I can think of is if there are custom icons in `traject` that are not in EDS, they'll need to pass those in via `children` anyway. But I think right now the only custom icon not in EDS is the LP logo, which I don't think is being used in `Button`...